src/borg/remote.py

```diff
@@ -53,6 +55,52 @@ class UnexpectedRPCDataFormatFromServer(Error):
     """Got unexpected RPC data format from server."""
 
 
+# Protocol compatibility:
+# In general the server is responsible for rejecting too old clients and the client it responsible for rejecting
+# too old servers. This ensures that the knowleadge what is compatible is always held by the newer component.
+#
+# The server can do checks for the client version in RepositoryServer.negotiate. If the client_data is 2 then then
+# client is in the version range [0.29.0, 1.0.6] inclusiv. For newer clients client_data is a dict which contains
+# client_version.
+#
+# For the client the return of the negotiate method is either 2 if the server is in the version range [0.29.0, 1.0.6]
+# inclusiv, or it is a dict which includes the server version.
+#
+# All method calls on the remote repository object must be whitelisted in RepositoryServer.rpc_methods and have api
+# stubs in RemoteRepository. The @api decorator on these stubs is used to set server version requirements.
+#
+# Method parameters are identfied only by name and never by position. If not overriden by a version requirement,
+# unknown parameters are ignored by the server side. When parameters are removed, they need to be preserved
+# as defaulted parameters either on the client stubs so that older servers still get the from them still needed parameters.
+
+
+compatMap = {
+    'check' : ['repair', 'save_space'],
+    'commit': ["save_space"],
+    'rollback': [],
+    'destroy': [],
+    '__len__': [],
+    'list': ["limit", "marker"],
+    'put': ["id_", "data"],
+    'get': ["id_"],
+    'delete': ["id_"],
+    'save_key': ["keydata"],
+    'load_key': [],
+    'break_lock': [],
+    'negotiate': ['client_data'],
+    'open': ['path', 'create', 'lock_wait', 'lock', 'exclusive', 'append_only'],
+    'get_free_nonce': [],
+    'commit_nonce_reservation': ['next_unreserved', 'start_nonce']
```

Comment:
please use single-quotes for string literals. do a global search within your changeset.